docs/decisions/006-ha-distributed-query.md

```diff
@@ -0,0 +1,267 @@
+# DR-006: High Availability Distributed Query with Stateless Schedulers
+
+## Status
+
+Draft
+
+## Context
+
+Following the decisions to adopt Apache Ballista as Spice's distributed query framework (DR-004) and extend it with catalog sync, UDF sync, and security features (DR-005), Spice needs to support production-grade high availability for distributed query execution.
+
+Stock Apache Ballista uses a single-scheduler model where the scheduler is a single point of failure. If the scheduler process dies:
+
+1. All in-flight queries are lost with no recovery mechanism.
+2. Clients must manually reconnect and resubmit queries.
+3. Executor state and task progress are orphaned.
+
+For production deployments, Spice requires:
+
+* **Active-active schedulers**: Multiple scheduler nodes behind a load balancer with no leader election or failover delays.
+* **Stateless schedulers**: Any scheduler can handle any request; scheduler pods may disappear/reappear at any time.
+* **Resilient query execution**: In-flight queries should fail gracefully when a scheduler dies, with clients able to resubmit to any scheduler.
```

Comment:
Shouldn't this be that in-flight queries are automatically retried? They shouldn't fail from the perspective of the client.

Having the in-flight queries fail is easier to implement and might simplify some of the job state we need to do, but I don't think that is what we want